Deriving macro doesn't work with `Result` alias

If a module uses a re-defined `Result` like the following code:

```rust
type Result<T> = std::result::Result<T, MyError>;
```

When you use `Nif` macros derivation below, it may occur the following error.

```rust
#[derive(NifMap)]
struct Test {
    a: i32,
}
```

```
error[E0107]: this type alias takes 1 generic argument but 2 generic arguments were supplied
 --> src/lib.rs:8:10
  |
8 | #[derive(rustler::NifMap)]
  |          ^^^^^^^^^^^^^^^
  |          |
  |          expected 1 generic argument
  |          help: remove this generic argument
  |
note: type alias defined here, with 1 generic parameter: `T`
 --> src/lib.rs:6:6
  |
6 | type Result<T> = std::result::Result<T, MyError>;
  |      ^^^^^^ -
```

Repo: https://github.com/SeokminHong/rustler-iss-result

--- a/rustler_codegen/src/ex_struct.rs
+++ b/rustler_codegen/src/ex_struct.rs
@@ -96,7 +96,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
 
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #struct_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 use #atoms_module_name::*;
                 use ::rustler::Encoder;
 
@@ -106,7 +106,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
                     env: rustler::Env<'a>,
                     term: rustler::Term<'a>,
                     field: rustler::Atom,
-                    ) -> Result<T, rustler::Error>
+                    ) -> ::rustler::NifResult<T>
                     where
                         T: rustler::Decoder<'a>,
                     {

--- a/rustler_codegen/src/map.rs
+++ b/rustler_codegen/src/map.rs
@@ -78,7 +78,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
 
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #struct_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 use #atoms_module_name::*;
 
                 let env = term.get_env();
@@ -87,7 +87,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
                     env: rustler::Env<'a>,
                     term: rustler::Term<'a>,
                     field: rustler::Atom,
-                    ) -> Result<T, rustler::Error>
+                    ) -> ::rustler::NifResult<T>
                     where
                         T: rustler::Decoder<'a>,
                     {

--- a/rustler_codegen/src/record.rs
+++ b/rustler_codegen/src/record.rs
@@ -99,7 +99,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
     };
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #struct_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 use #atoms_module_name::*;
 
                 let terms = match ::rustler::types::tuple::get_tuple(term) {
@@ -118,7 +118,7 @@ fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> T
                     return Err(::rustler::Error::RaiseAtom("invalid_record"));
                 }
 
-                fn try_decode_index<'a, T>(terms: &[::rustler::Term<'a>], pos_in_struct: &str, index: usize) -> Result<T, rustler::Error>
+                fn try_decode_index<'a, T>(terms: &[::rustler::Term<'a>], pos_in_struct: &str, index: usize) -> ::rustler::NifResult<T>
                     where
                         T: rustler::Decoder<'a>,
                 {

--- a/rustler_codegen/src/tuple.rs
+++ b/rustler_codegen/src/tuple.rs
@@ -83,13 +83,13 @@ fn gen_decoder(ctx: &Context, fields: &[&Field]) -> TokenStream {
     };
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #struct_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 let terms = ::rustler::types::tuple::get_tuple(term)?;
                 if terms.len() != #field_num {
                     return Err(::rustler::Error::BadArg);
                 }
 
-                fn try_decode_index<'a, T>(terms: &[::rustler::Term<'a>], pos_in_struct: &str, index: usize) -> Result<T, rustler::Error>
+                fn try_decode_index<'a, T>(terms: &[::rustler::Term<'a>], pos_in_struct: &str, index: usize) -> ::rustler::NifResult<T>
                     where
                         T: rustler::Decoder<'a>,
                 {

--- a/rustler_codegen/src/unit_enum.rs
+++ b/rustler_codegen/src/unit_enum.rs
@@ -87,7 +87,7 @@ fn gen_decoder(ctx: &Context, variants: &[&Variant], atoms_module_name: &Ident) 
 
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #enum_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 use #atoms_module_name::*;
 
                 let value = ::rustler::types::atom::Atom::from_term(term)?;

--- a/rustler_codegen/src/untagged_enum.rs
+++ b/rustler_codegen/src/untagged_enum.rs
@@ -67,7 +67,7 @@ fn gen_decoder(ctx: &Context, variants: &[&Variant]) -> TokenStream {
 
     let gen = quote! {
         impl<'a> ::rustler::Decoder<'a> for #enum_type {
-            fn decode(term: ::rustler::Term<'a>) -> Result<Self, ::rustler::Error> {
+            fn decode(term: ::rustler::Term<'a>) -> ::rustler::NifResult<Self> {
                 #(#variant_defs)*
 
                 Err(::rustler::Error::RaiseAtom("invalid_variant"))

